CMakeLists.txt

```diff
@@ -156,6 +156,13 @@ endif()
 zephyr_compile_options($<$<COMPILE_LANGUAGE:C>:$<TARGET_PROPERTY:compiler,no_strict_aliasing>>)
 zephyr_compile_options($<$<COMPILE_LANGUAGE:CXX>:$<TARGET_PROPERTY:compiler-cpp,no_strict_aliasing>>)
 
+# Extra warnings options for twister run
+if (FORCE_WARNING_AS_ERRORS)
```

Comment:
Why is a CMake variable chosen here, and not for example a Kconfig setting ?

Initial feeling is this probably should have a Kconfig setting, but let me know if there are good reasons not to do so.

Reply:
CMakeLists.txt sets compiller arguments templates regarding FORCE_WARNING_AS_ERRORS variable value.
The variable itself is initially configured in twister script and it is required for run tests only. I'm not pretty sure it is correct to move tests-related settings into some general Kconfig.

Reply:
Thanks for your reason.

Then I believe a proper Kconfig setting for this should be created.
This in order to be consistent with how we generally handle build settings (including compiler / linker configuration).

It will also make better support for release / debug targets cleaner in future.

The old style `EXTRA_xFLAGS` are generic and pass anything stright to the toolchain, but the new approach has pre-defined bahvoir and thus Kconfig should be used.
Kconfig help text also serves as documentation of the setting and are rendered into Zephyr docs.